template/node12/index.js

```diff
@@ -61,23 +63,29 @@ function handleAppResource(req, res) {
     }
 }
 
+async function initManifest() {
+    if (manifest == null) {
+        let possibleFutureRes = manifestHandler();
+        let tempManifest = await possibleFutureRes;
+        manifest = {};
+        widgetHandlers = tempManifest.widgets;
+        listenerHandlers = tempManifest.listeners || {};
+        manifest.widgets = Object.keys(widgetHandlers);
+        manifest.listeners = Object.keys(listenerHandlers);
+        manifest.rootWidget = tempManifest.rootWidget;
```

Comment:
Can be simplified by : 
```suggestion
        widgetHandlers = tempManifest.widgets;
        listenerHandlers = tempManifest.listeners || {};
        manifest = {
          widgets: Object.keys(widgetHandlers),
          listeners: Object.keys(listenerHandlers),
          rootWidget: tempManifest.rootWidget
        };
```